### PR TITLE
[WIP] Tooltip title reflects binned date

### PR DESCRIPTION
**Pull request by clhenrick**
Relates to https://github.com/mapd/mapd-immerse/issues/3746

When binning a date, the tooltip title should show the corresponding date type, eg. for "month" it should show the month name instead of an entire date string.

### CODE REVIEW

**Comment by clhenrick** [comment]
@biovisualize I think this is ready for merge with `develop`, peer review, S'il vous plaît?

File: src/charts/tooltip.js [hunk 1]
```diff
@@ -236,9 +246,35 @@ export default function Tooltip (_container, isLegend = false) {
...
+    if (title instanceof Date) {
+      const { binningResolution } = config;
+      let specifier = binTranslation[binningResolution]
+
+      if (specifier) {
+        title = d3.utcFormat(specifier)(title)
+      } else if (binningResolution === "1q") {
+        // handle bin translation for bin types not available in d3-time (century, decade, quarter)
+        // currently only handling "1q"
+        title = formatOddDateBin(binningResolution, title, data)
```
Comment (biovisualize) [inline src/charts/tooltip.js]: Why do you need data here if you don't use it in formatOddDateBin?

File: src/charts/tooltip.js [hunk 2]
```diff
@@ -297,6 +333,11 @@ export default function Tooltip (_container, isLegend = false) {
     return this
   }
 
+  function setData (_data) {
```
Comment (biovisualize) [inline src/charts/tooltip.js]: That would not be needed depending on your answer on my previous comment.

File: src/charts/tooltip.js (hunk 1 above)
Comment (clhenrick) [reply src/charts/tooltip.js]: my bad, I thought I had removed that last arg

File: src/charts/tooltip.js (hunk 2 above)
Comment (clhenrick) [reply src/charts/tooltip.js]: right, I was originally going to try to replicate Immerse's decade binning behavior, in which case would need the data object, but then we decided against it. The data object might come in handy in the future for the tooltip which is why I left that method intact. I'm okay with removing it though.

**Comment by clhenrick** [comment]
@biovisualize updates made

**Comment by biovisualize** [comment]
Thanks! I'm merging it. We can polish later if needed. I will flag QA on the related issue. They can test it on the combo-chart PR

**Pull request merged**